### ISSUE TITLE
BitBucket branches containing forward slash "/" are unaccessable

### ISSUE THREAD

**Issue by endre-seqera** [nextflow-io/nextflow#4599]
### Expected behavior and actual behavior

Nextflow should be able to execute pipelines stored on BitBucket repository, under a branch name containing forward slash "/".

### Steps to reproduce the problem

To ease testing, i've created a public [bitbucket repo with nextflow-hello,](https://bitbucket.org/endre-seqera-testing/nextflow-hello/src/e12ed7715dd91991b31bd848070d4736d4997ec7/?at=feature%2F1234), with two branches `master` and `feature/1234`.

When trying to run the pipeline from branch `feature/1234`, we get an error.
```
nextflow run https://bitbucket.org/endre-seqera-testing/nextflow-hello -r feature/1234
N E X T F L O W  ~  version 23.10.0
Pulling endre-seqera-testing/nextflow-hello ...
WARN: Cannot read project manifest -- Cause: Remote resource not found: https://bitbucket.org/api/2.0/repositories/endre-seqera-testing/nextflow-hello/src/feature/1234/nextflow.config
Remote resource not found: https://bitbucket.org/api/2.0/repositories/endre-seqera-testing/nextflow-hello/src/feature/1234/main.nf
```

If we run from the `master` branch, the execution succeeds:
```
nextflow run https://bitbucket.org/endre-seqera-testing/nextflow-hello -r master
N E X T F L O W  ~  version 23.10.0
Pulling endre-seqera-testing/nextflow-hello ...
 downloaded from https://bitbucket.org/endre-seqera-testing/nextflow-hello.git
Launching `https://bitbucket.org/endre-seqera-testing/nextflow-hello` [silly_keller] DSL2 - revision: e12ed7715d [master]
executor >  local (4)
[bf/6df11c] process > sayHello (1) [100%] 4 of 4 ✔
Hola world!

Hello world!

Ciao world!

Bonjour world!
```
> [!NOTE]  
> After one successful execution from the `master` branch, the repo has been downloaded (cached locally), and subsequent executions work even from the `feature/1234`. However, if we clear the cache `~/.nexflow/assets/endre-seqera-testing` the error returns.

### Program output 

N/A

### Environment 

* Nextflow version: 23.10.0 build 5889
* Java version: openjdk 21.0.1 2023-10-17 LTS
* Operating system: Mac OS X 14.1.2
* Bash version: zsh 5.9 (x86_64-apple-darwin23.0)

### Additional context

As mentioned in [the original issue opened](https://github.com/seqeralabs/nf-tower-cloud/issues/6055#issuecomment-1856274905) this is a know bug in BitBucket. But a workaround could be implemented using and additional API call.

Simple curl commands can be used to reproduce as well:
```
curl https://bitbucket.org/api/2.0/repositories/endre-seqera-testing/nextflow-hello/src/feature/1234/nextflow.config
{"type":"error","error":{"message":"Commit not found","data":{"shas":["feature"]}},"data":{"shas":["feature"]}}%
```

Possible workaround, using commit ID (which can be retrieved by another API call to /branches) instead of branch name:
```
curl https://bitbucket.org/api/2.0/repositories/endre-seqera-testing/nextflow-hello/src/e12ed7715dd91991b31bd848070d4736d4997ec7/nextflow.config
process.container = 'quay.io/nextflow/bash'
```

After some investigation, the code that fails is in the `BitbucketRepositoryProvider.readBytes()` method. Adding an additional unit test, can reproduce it as well in the `BitbucketRepositoryProviderTest`:
```
    def 'readText should work with different branches' () {
        given:
        String CONFIG = '''
        providers {
            mybitbucket {
                server = 'https://bitbucket.org'
                endpoint = 'https://bitbucket.org'
                platform = 'bitbucket'
            }
        }
        '''

        def config = new ConfigSlurper().parse(CONFIG)
        def obj = new ProviderConfig('bitbucket', config.providers.mybitbucket as ConfigObject)

        expect:
        def bitBucketProvider = new BitbucketRepositoryProvider('endre-seqera-testing/nextflow-hello', obj)
        bitBucketProvider.setRevision('master')
        bitBucketProvider.readText('nextflow.config') == "process.container = 'quay.io/nextflow/bash'\n"

        bitBucketProvider.setRevision('feature/1234')
        // will fail now
        bitBucketProvider.readText('nextflow.config') == "process.container = 'quay.io/nextflow/bash'\n"
    }
```

**Comment by endre-seqera** [comment]
fix was merged [5893783](https://github.com/nextflow-io/nextflow/commit/58937831ee789c32160c79c94ace97ff89e322ce)

**Comment by endre-seqera** [comment]
and released: [v23.12.0-edge](https://github.com/nextflow-io/nextflow/releases/tag/v23.12.0-edge)
